### ISSUE TITLE
[YYC] Allocating tens of thousands of arrays per frame causes memory leak

### ISSUE THREAD

**Issue by thennothinghappened** [YoYoGames/GameMaker-Bugs#5677]
### Description
This is an issue based off a discussion about array allocation freezing the runner in the beta chat. We did some testing including YYC and on my system at least, YYC doesn't seem to reclaim memory properly when so many things are allocated. As it ran, it consumed all of the available memory and then silently crashed.

### Steps To Reproduce
This is the sample step event from the original discussion, not my creation:

```gml
var _time = current_time;
var _count = 0;

for (var _i = 0; _i < 90000; _i++)
{
    debug_test_function();
	_count ++;
}

show_debug_message($"test completed in {current_time -_time} ms. {_count} calls."); 
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.400.0.532 Runtime v2024.400.0.551

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

[55249c3a-6e01-4f9a-b5c5-81361febca23](https://api.gamemaker.io/api/github/downloads/55249c3a-6e01-4f9a-b5c5-81361febca23)

**Comment by YYDan** [comment]
Yes, apparently internally it has also been confirmed to be VM only. (YYC is apparently just an issue with Task Manager not reporting the memory dropping back down as quickly as it actually happens.)

**Comment by rwkay** [comment]
This is the PR for the fix - https://github.com/YoYoGames/GameMaker/pull/3951

**Comment by yymrennie** [comment]
Fixed in 2024.4

**Comment by caitlinrmcintyre** [comment]
Verified fix in Beta IDE v2024.400.0.556  Runtime v2024.400.0.571, thanks.
